addons/stock/report/report_stock_forecasted.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from collections import defaultdict
+
+from odoo import api, models
+from odoo.tools import float_is_zero, format_datetime, format_date
+
+
+class ReplenishmentReport(models.AbstractModel):
+    _name = 'report.stock.report_product_product_replenishment'
+    _description = "Stock Replenishment Report"
+
+    def _product_domain(self, product_template_ids, product_variant_ids):
+        if product_template_ids:
+            return [('product_tmpl_id', 'in', product_template_ids)]
+        return [('product_id', 'in', product_variant_ids)]
+
+    def _move_domain(self, product_template_ids, product_variant_ids, wh_location_ids):
+        move_domain = self._product_domain(product_template_ids, product_variant_ids)
+        move_domain += [('product_uom_qty', '!=', 0)]
+        out_domain = move_domain + [
+            '&',
+            ('location_id', 'in', wh_location_ids),
+            ('location_dest_id', 'not in', wh_location_ids),
+        ]
+        in_domain = move_domain + [
+            '&',
+            ('location_id', 'not in', wh_location_ids),
+            ('location_dest_id', 'in', wh_location_ids),
+        ]
+        return in_domain, out_domain
+
+    def _move_draft_domain(self, product_template_ids, product_variant_ids, wh_location_ids):
+        in_domain, out_domain = self._move_domain(product_template_ids, product_variant_ids, wh_location_ids)
+        in_domain += [('state', '=', 'draft')]
+        out_domain += [('state', '=', 'draft')]
+        return in_domain, out_domain
+
+    def _move_confirmed_domain(self, product_template_ids, product_variant_ids, wh_location_ids):
+        in_domain, out_domain = self._move_domain(product_template_ids, product_variant_ids, wh_location_ids)
+        out_domain += [('state', 'not in', ['draft', 'cancel', 'done'])]
+        in_domain += [('state', 'not in', ['draft', 'cancel', 'done'])]
+        return in_domain, out_domain
+
+    def _compute_draft_quantity_count(self, product_template_ids, product_variant_ids, wh_location_ids):
+        in_domain, out_domain = self._move_draft_domain(product_template_ids, product_variant_ids, wh_location_ids)
+        incoming_moves = self.env['stock.move'].read_group(in_domain, ['product_qty:sum'], 'product_id')
+        outgoing_moves = self.env['stock.move'].read_group(out_domain, ['product_qty:sum'], 'product_id')
+        in_sum = incoming_moves[0]['product_qty'] if incoming_moves else 0
```

Comment:
If you have multiple variant. It will only take the quantity for the first one no?

Reply:
Indeed... I changed with that:
```python
in_sum = sum(move['product_qty'] for move in incoming_moves)
out_sum = sum(move['product_qty'] for move in outgoing_moves)
```